### DISCUSSION TITLE
Set client ssl certificate

### DISCUSSION THREAD

**Discussion by albertovincenzi** [php-webdriver/php-webdriver#840] [Q&A] [answered]
### What are you trying to achieve? (Expected behavior)
I need to specify a client certificate to connect to a specific host. I must discard any chrome solution because the website I need to scrape doesn't support it (very strange but true). With phantomjs I found the option to pass the certificate but seems to be a discontinued solution (and I wasn't able to make phantomjs work with latest version).
htmlunit have the option setSSLClientCertificate but you can't set that using setCapability()
firefox is quite hard to configure as I think that the only solution is to create a profile every time I run a new session (I must support multiple certificates)

Do you have any idea? I wasn't able to find any documentation related to my issue

**Comment by OndraM** [comment, marked as answer]
Well, PhantomJS is deprecated. It will be quite hard to make some page fully working in HtmlUnit. So I will go the Firefox path. 

Creating profile or reusing existing one is not difficult:
- Either use `FirefoxProfile()` and put it in Capabiltities when creating new browser driver
- or pass `--profile /path/to/existing/profile` as an `args` capability of Firefox - see https://firefox-source-docs.mozilla.org/testing/geckodriver/Profiles.html .

This will be something like:
```php
$capabilities = DesiredCapabilities::firefox();
$capabilities->setCapability(
    'moz:firefoxOptions',
    ['args' => ['--profile /profile/location']]
);

$driver = RemoteWebDriver::create($seleniumServerHost, $capabilities);
```

**Comment by albertovincenzi** [comment]
thanks @OndraM,
I read about the solution you proposed but I had to make the certificate selection dynamic.
I ended up making the scraper with the traditional way (goutte) which for such old websites is even better. Hopefully, it will be very hard in future to have modern website (which require headless browser) that will implement such authentication method.
